halo2-wasm/src/halo2lib.rs

```diff
@@ -328,6 +329,77 @@ impl Halo2LibWasm {
         self.to_js_assigned_values(out)
     }
 
+    pub fn to_hi_lo(&mut self, a: usize) -> Vec<u32> {
+        let a = self.get_assigned_value(a);
+        let a_val = a.value();
+        let a_bytes = a_val.to_bytes();
+
+        let mut a_lo_bytes = [0u8; 32];
+        let mut a_hi_bytes = [0u8; 32];
+        a_lo_bytes[..16].copy_from_slice(&a_bytes[..16]);
+        a_hi_bytes[..16].copy_from_slice(&a_bytes[16..]);
+        let a_lo = Fr::from_bytes(&a_lo_bytes).unwrap();
+        let a_hi = Fr::from_bytes(&a_hi_bytes).unwrap();
+
+        let a_lo = self.builder.borrow_mut().main(0).load_witness(a_lo);
```

Comment:
when I tried this, I get `already borrowed: BorrowMutError` when actually using `ctx` afterwards

Reply:
hmmm maybe it's because either `self.get_assigned_value` or `self.get_js_value` is also mutably borrowing and not releasing?
Sometimes you also have to manually call `drop(builder)` when you're done with the code block using it.

We can probably investigate later once the CLI tool is available for easier testing.